OO#434 limiting the number of participants depending on the number of…

mcmpp opened the pull request:
… participants already subscribed

mcmpp committed:
OO#434 limiting the number of participants depending on the number of participants already subscribed

mcmpp committed:
OO#434 syntax warning fixing

rawyler commented on app/scripts/arbeitsangebote/list/arbeitsangebot-participate.controller.js:
```diff
@@ -4,24 +4,51 @@
+          var ret = undefined;
+          angular.forEach($scope.arbeitseinsatzList, function(arbeitseinsatz) {
+            if (arbeitseinsatz.arbeitsangebotId === arbeitsangebot.id) {
+              ret = arbeitseinsatz;
```
So ret is going to be last. Could also break or return here, right?

mcmpp committed:
OO#434 remove unused fucntion

AlwinEgger commented on app/scripts/arbeitsangebote/list/arbeitsangebotelist.controller.js:
```diff
@@ -70,6 +70,15 @@ angular
       }
 
+      $scope.availableVacancies = function(arbeitsangebot) {
+        if ((arbeitsangebot.anzahlPersonen-arbeitsangebot.anzahlEingeschriebene) === 0) {
```
maybe `<= `0 in case the admin added more persons than the maximum?

AlwinEgger commented on app/scripts/arbeitseinsaetze/list/arbeitseinsaetzelist.html:
```diff
@@ -34,8 +34,14 @@
+                        Sie können diese Zuordnung nicht entfernen, da sie zu nahe am Ausführungsdatum liegt
+                      </span>
+                    </script>
+                    <a ng-if="(arbeitseinsatz.zeitBis.getTime()<limitDateForDeletion)" ng-disabled="(arbeitseinsatz.zeitBis.getTime()<limitDateForDeletion)" role="button" uib-tooltip-template="'workAssignmentInfoTooltipTemplate.html'" class="btn btn-sm btn-warning ng-scope" ng-click="(arbeitseinsatz.zeitBis.getTime()<limitDateForDeletion) || quit(arbeitseinsatz)"><span class="glyphicon glyphicon- ...
```
ng-if and ng-disabled is the same. Shouldn't it be on ng-disabled only? And then the uib-tooltip-template should only display the above message, if the button is disabled, right? This could be acheived on the template.

AlwinEgger commented on i18n/template.pot:
```diff
@@ -10,7 +10,7 @@ msgid "#&nbsp;Pers."
 
 #: app/scripts/arbeitsangebote/list/arbeitsangebotelist.html:31
-msgid "+"
+msgid "&nbsp;+"
```
Is this ok?

mcmpp committed:
OO#434 integrate pull request changes

mcmpp commented:
Hi @AlwinEgger 
Since I have push my change I cannot answer you to the last comment so I answer you here. I have added the ngif in order to be able to display or not display the message. I was trying to limit the message only in template but for some reason this didn't work. I have added the second button where the user can change the quantity of people

AlwinEgger commented on app/scripts/arbeitseinsaetze/list/arbeitseinsaetzelist.html:
```diff
@@ -34,8 +34,19 @@
+                      </span>
+                    </script>
+                    <div ng-if="(arbeitseinsatz.zeitBis.getTime()<limitDateForDeletion)">
+                      <a ng-disabled="(arbeitseinsatz.zeitBis.getTime()<limitDateForDeletion)" role="button" uib-tooltip-template="'workAssignmentInfoTooltipTemplate.html'" class="btn btn-sm btn-warning ng-scope" ng-click="(arbeitseinsatz.zeitBis.getTime()<limitDateForDeletion) || quit(arbeitseinsatz)"><span class="glyphicon glyphicon-minus"></span></a>
```
I still don't see the need for the ng-disabled here. It will always be disabled, no? Since it's the same statement as in the ng-if.
Sorry for being picky.
